services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java

```diff
@@ -81,7 +86,7 @@ private String resolveHost(SdkHttpRequest request, String accountId, S3ControlCo
 
         }
         validateComponentIsHostnameCompliant(accountId, "account id");
-        return String.format("%s.%s", accountId, host);
+        return hasEndpointOverride ? host : String.format("%s.%s", accountId, host);
```

Comment:
Thinking about this some more, I'm not sure that this is the right behavior; it's possible that the user could still want this behavior even with an endpoint override.

Can we leave this change out of the client for now? For testing, we can we can just add a new client interceptor that changes the request URL to the WireMock host before it's sent.

Reply:
Yes, that sounds like a less intrusive change.